### PR TITLE
Improve Figure docstrings

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

The title says all.

### CODE REVIEW

File: lib/matplotlib/colorbar.py [hunk 1]
```diff
@@ -152,7 +152,7 @@
     default to the current image.
 
 cax : :class:`~matplotlib.axes.Axes` object, optional
-    Axis into which the colorbar will be drawn
+    Axis into which the colorbar will be drawn.
```
Comment (jklymak) [inline lib/matplotlib/colorbar.py]: Should be "Axes"

**Comment by timhoffm** [comment]
Strange. Just updated a typo in a docstring and suddenly an image comparison test fails. Did anything else in the build environment change?
Note that the rms 0.011 is only slightly larger than the given tolerance 0.1 for the test.

**Comment by jklymak** [comment]
Yeah, I've seen the same thing.  Restarting the job often gets it to pass.  Not sure what causes the instability.

**Comment by timhoffm** [comment]
Is there a way to restart just a specific CI service or job? Up to now, I've usually rebased in such a case, which then triggers the CI as a side effect.

**Comment by jklymak** [comment]
For Travis, you can just hit restart.

For appveyor I think you need to log in and can restart.  Its a bit of a pain.

**Comment by QuLogic** [comment]
The failures are consistent across tests, but I'm not sure why they suddenly started appearing.

**Comment by timhoffm** [comment]
From the changed rms I would assume that some commit slightly changed the output, just about enough to make this test fail.

**Comment by timhoffm** [comment]
Rebased.

File: lib/matplotlib/figure.py [hunk 2]
```diff
@@ -462,7 +459,12 @@ def show(self, warn=True):
     def _get_axes(self):
         return self._axstack.as_list()
 
-    axes = property(fget=_get_axes, doc="Read-only: list of axes in Figure")
+    axes = property(fget=_get_axes,
+                    doc="List of axes in the Figure. You can access and "
+                        "modify the axes in the Figure through this list. "
+                        "Do not modify the list itself. Instead, use "
```
Comment (anntzer) [inline lib/matplotlib/figure.py]: the second and third sentence are contradicting each other.

File: lib/matplotlib/figure.py [hunk 3]
```diff
@@ -935,40 +939,40 @@ def get_facecolor(self):
...
         return self.bbox_inches.height
 
     def get_dpi(self):
-        """Return the dpi as a float."""
+        """Return the resolution in dots per inch as a float."""
         return self.dpi
 
     def get_frameon(self):
-        """Get the boolean indicating frameon."""
+        """Return a bool indicating whether the figure frame will be dawn."""
```
Comment (anntzer) [inline lib/matplotlib/figure.py]: I tend to write these just as "return whether ..." which seems less verbose and just as clear.
Also typo: dawn.

File: lib/matplotlib/figure.py (hunk 2 above)
Comment (timhoffm) [reply lib/matplotlib/figure.py]: No. Assume `axs = fig.axes` being `[ax1, ax2]`. You can access and modify `ax1`, e.g. axs[0].set_xlim(0, 1)`. But you cannot add or remove an axes from the figure by `fig.axes.append(ax3)` or similar.

Suggestions how to describe this more clearly are welcome.

Comment (anntzer) [reply lib/matplotlib/figure.py]: just remove the "and modify" in the first sentence?  it's not as if python had a notion of const reference anyways...

**Comment by anntzer** [comment]
thanks

**Pull request merged**